src/ir_def/ir_effect.ml

```diff
@@ -20,12 +20,29 @@ let is_triv phrase = eff phrase = T.Triv
 
 let effect_exp (exp: exp) : T.eff = eff exp
 
+let is_async_call p exps =
+  match (p, exps) with
+  | CallPrim _, [exp1; _] ->
+    T.is_shared_func (typ exp1) ||
+    T.is_local_async_func (typ exp1)
```

Comment:
I assume this also includes fire&forget sends?

Reply:
Yes, it should. We only have remote fire-and-forget and both remote request and remote oneways are shared funcs.